### DISCUSSION TITLE
TS type issues using dependent queries not working with prefetchQuery

### DISCUSSION THREAD

**Discussion by waptik** [TanStack/query#1596] [Q&A] [answered]
It seems `enabled` is not part of  `FetchQueryOptions` when using `QueryClient.prefetchQuery`, same with `useQuery`.
This is what intellisense show:
```ts
Argument of type '{ enabled: boolean; }' is not assignable to parameter of type 'FetchQueryOptions<unknown, unknown, unknown>'.
  Object literal may only specify known properties, and 'enabled' does not exist in type 'FetchQueryOptions<unknown, unknown, unknown>'.ts(2345)
```

Sample prefetch code
```ts
...
const isSinglePost = ...
await queryClient.prefetchQuery(
			[CREATORS_QUERIES.getPosts, { creator }],
			() => {
                   console.log("hello");
				//return;
			},
			{
				enabled: isSinglePost,
			},
		);
```

i'm using version 3.5.11

**Comment by TkDodo** [comment, marked as answer]
`prefetchQuery` is a side-effectful method that should not be called during render. If you call it in an effect or in an event handler or something like that, you can just add an if statement around it and _not_ call it at all. 

So, there is no enabled option on prefetchQuery.

**Reply by waptik** [reply]
I'm using it inside nextjs' `getStaticProps`. After opening this discussion issue, i wrapped it inside an if statement statement and removed the `enabled` option. That fixed the issue.

I suppose that should be made clear in the docs at https://react-query.tanstack.com/reference/QueryClient#queryclientprefetchquery where it states that `The options for prefetchQuery are exactly the same as those of useQuery.`

**Reply by TkDodo** [reply]
> I'm using it inside nextjs' getStaticProps. After opening this discussion issue, i wrapped it inside an if statement statement and removed the enabled option. That fixed the issue.

👍 

> where it states that The options for prefetchQuery are exactly the same as those of useQuery.

yes, good point. The type definitions are always the source of truth because they are derived from source code, since the lib is written in TypeScript. The types for `prefetchQuery` are here:
https://github.com/tannerlinsley/react-query/blob/361935a12cec6f36d0bd6ba12e84136c405047c5/src/core/queryClient.ts#L285-L289
and `FetchQueryOptions` are here:
https://github.com/tannerlinsley/react-query/blob/361935a12cec6f36d0bd6ba12e84136c405047c5/src/core/types.ts#L196-L200

if you want, you can open a PR to improve the docs. I would maybe say that the options for `prefetchQuery` are exactly the same as those of `fetchQuery`, and then describe in `fetchQuery` what those options are, or at least how they differ from `useQuery`. And then of course there's also `(pre)fetchInfiniteQuery`, which is slightly different.

**Reply by waptik** [reply]
I created a PR(#1609) to state the options not available to the prefetch method.
